lib/cli.js

```diff
@@ -29,6 +29,17 @@ const debug = require("debug")("eslint:cli");
 //------------------------------------------------------------------------------
 
 /**
+ * Predicate function for whether or not to apply fixes in quiet mode.
+ * If a message is a warning, do not apply a fix.
+ * @param {LintResult} lintResult The lint result.
+ * @returns {boolean} True if the lint message is an error (and thus should be
+ * autofixed), false otherwise.
+ */
+function quietFixPredicate(lintResult) {
+    return Boolean(lintResult && lintResult.severity === 2);
```

Comment:
It probably should never be null. Normally I favor defensive coding to be safe; but since this could be a tight path (running this function on every lint message), I am happy to shave off those cycles. I'll update tonight (US).